### ISSUE TITLE
Drop actual_path?

### ISSUE THREAD

**Issue by Arcterus** [mesalock-linux/mesabox#33]
I am considering removing `util::actual_path()` and just making all the utilities assume that the current working directory that we should operate in is the current working directory of the binary.  The original point of `util::actual_path()` was to support the testing framework, but we won't need it for that once #24 is merged.

I can see it still being useful in some cases where `mesabox` is used as a library within another program as it allows operating in different directories without modifying the arguments or changing the host program's current working directory (which could be problematic if that program were to be multi-threaded).  At the same time, removing `util::actual_path()` makes the utilities simpler and reduces the number of allocations we need to perform.

**Comment by Arcterus** [comment]
If no one has a preference/feels like it is useful, I am leaning towards dropping it as I'm not sure the added complexity is worth it.

**Comment by Arcterus** [comment]
Hit the wrong button.

**Comment by mssun** [comment]
No preference for me. I think we can drop it to keep the the code clean and simple.

**Comment by Arcterus** [comment]
Only thing about this that needs consideration really is what to do with `cd` in `sh`.  Currently, we try to avoid forking as much as possible, so we don't actually spawn real subshells if we can avoid doing so.  The problem with that is that if the user changes the directory in one of the fake "subshells," the directory will changed outside of the "subshell."

My line of thought right now is that we should avoid forking until we see `cd`, in which case we fork and execute the rest of the commands for the current subshell in a _real_ subshell.  Doing so requires keeping track of whether we are in a fake "subshell" and ensuring we only fork when we haven't already done so for the current "subshell."

Another solution might be to change the directory and then change it back once the "subshell" exits, but this could cause issues if `env::set_current_dir()` fails after the current working directory has already been changed.  This solution might also cause problems if we ever add multi-threading or other related features.
